src/lib/candid_types/func.ts

```diff
@@ -1,8 +1,59 @@
-import { Principal } from './';
+import { Principal, Service } from './';
 
-export type Func<T> = T;
+/**
+ * Represents a reference to a {@link Service} method. The type param specifies
+ * the functional signature of the method but under the hood it is really just
+ * an array with two values:
```

Comment:
I think we should explain that it's a TypeScript tuple, it's not really an array. Well the type is a TypeScript tuple, at runtime it will be an array. I think mentioning the true TypeScript type would be beneficial.